src/App.tsx

```diff
@@ -17,18 +55,31 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setStatus={setStatus}
+                setQuery={setQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isDataLoaded ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={filteredTodos}
+                  setTodo={setSelectedTodo}
+                  todoId={selectedTodo ? selectedTodo.id : undefined}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal todo={selectedTodo} onClose={setSelectedTodo} />
```

Comment:
The `onClose` prop for `TodoModal` should be a function that sets `selectedTodo` to `null`. Instead of passing `setSelectedTodo` directly, you can pass a function like `() => setSelectedTodo(null)` to ensure the modal closes as expected.